clientv3/auth.go

```diff
@@ -41,8 +43,11 @@ type Auth interface {
 	// UserChangePassword changes a password of a user.
 	UserChangePassword(ctx context.Context, name string, password string) (*AuthUserChangePasswordResponse, error)
 
-	// RoleAdd adds a new user to an etcd cluster.
+	// RoleAdd adds a new role to an etcd cluster.
 	RoleAdd(ctx context.Context, name string) (*AuthRoleAddResponse, error)
+
+	// RoleGrant grants a permission to a role.
+	RoleGrant(ctx context.Context, name string, key string, permType authpb.Permission_Type) (*AuthRoleGrantResponse, error)
```

Comment:
let's type define authpb.Permission_Type in this client pkg. basically, we do not want to expose any external types in client library.